src/java.desktop/share/classes/javax/sound/midi/Track.java

```diff
@@ -199,9 +199,7 @@ public boolean remove(MidiEvent event) {
         // by deleting events (unless the EOT event is removed)
         synchronized(eventsList) {
             if (set.remove(event)) {
-                int i = eventsList.indexOf(event);
-                if (i >= 0) {
-                    eventsList.remove(i);
+                if (eventsList.remove(event)) {
```

Comment:
Can be rewritten as `if (set.remove(event) && eventsList.remove(event))`.

Reply:
`return set.remove(event) && eventsList.remove(event);` even...

Reply:
This is *cleaner*.